src/main/java/com/codurance/retropolis/controllers/BoardController.java

```diff
@@ -14,14 +20,20 @@
 public class BoardController extends BaseController {
 
   private final BoardService boardService;
+  private final GoogleTokenAuthenticator tokenAuthenticator;
+  private final UserService userService;
 
   @Autowired
-  public BoardController(BoardService boardService) {
+  public BoardController(BoardService boardService, GoogleTokenAuthenticator tokenAuthenticator, UserService userService) {
     this.boardService = boardService;
+    this.tokenAuthenticator = tokenAuthenticator;
+    this.userService = userService;
   }
 
   @GetMapping(value = "/{id}")
-  public Board getBoard(@PathVariable Long id) {
+  public Board getBoard(@PathVariable Long id, @RequestHeader(HttpHeaders.AUTHORIZATION) String token)
+      throws GeneralSecurityException, IOException {
+    userService.registerUserIfNotExists(tokenAuthenticator.getEmail(token), id);
```

Comment:
i'm not sure about the name of this. it doesn't just register a user if not exists, it also adds them to a board right? the userService `findOrCreate` does the register if not exists right? and i've made that user one public in my branch to make it reusable for `/boards`

Reply:
Any suggestions for the function name?
Yes, the `findOrCreate` will only register if the user does not exists